consul/src/main/scala/io/buoyant/consul/v1/AgentApi.scala

```diff
@@ -0,0 +1,40 @@
+package io.buoyant.consul.v1
+
+import com.twitter.conversions.time._
+import com.twitter.finagle.http
+import com.twitter.finagle.service.Backoff
+import com.twitter.finagle.stats.{DefaultStatsReceiver, StatsReceiver}
+import com.twitter.util.{Closable, Duration, Future}
+
+object AgentApi {
+  def apply(c: Client): AgentApi = new AgentApi(c, s"/$versionString")
+}
+
+class AgentApi(
+  val client: Client,
+  val uriPrefix: String,
+  val backoffs: Stream[Duration] = Backoff.exponentialJittered(1.milliseconds, 5.seconds),
+  val stats: StatsReceiver = DefaultStatsReceiver
+) extends BaseApi with Closable {
+
+  val agentPrefix = s"$uriPrefix/agent"
+
+  // https://www.consul.io/docs/agent/http/agent.html#agent_self
+  def self(retry: Boolean = false): Future[Self] = {
+    val req = mkreq(http.Method.Get, s"$agentPrefix/self")
+    executeJson[Self](req, retry).map(_.value)
+  }
+}
+
+// Classes below are used for JSON parsing hence field names case have to match the on in JSON
+
+// Represents configuration and member information of the local agent
+// For now only Config is parsed as other fields are unused
+case class Self(
```

Comment:
What do you think about calling this `LocalAgent` or something like that?  `Self` (and `self` above) is a bit confusing because of scala's self-types.